middleware/security/jwt/jwt.go

```diff
@@ -134,6 +137,30 @@ func New(resolver KeyResolver, validationFunc goa.Middleware, scheme *goa.JWTSec
 	}
 }
 
+func extractTokenFromHeader(schemeName string, req *http.Request) (string, error) {
+	val := req.Header.Get(schemeName)
+	if val == "" {
+		return "", ErrJWTError(fmt.Sprintf("missing header %q", schemeName))
+	}
+
+	if !strings.HasPrefix(strings.ToLower(val), "bearer ") {
+		return "", ErrJWTError(fmt.Sprintf("invalid or malformed %q header, expected 'Bearer JWT-token...'", val))
+	}
+
+	incomingToken := strings.Split(val, " ")[1]
+
+	return incomingToken, nil
+}
+
+func extractTokenFromQueryParam(schemeName string, req *http.Request) (string, error) {
+	incomingToken := req.URL.Query().Get(schemeName)
+	if incomingToken == "" {
+		return "", ErrJWTError(fmt.Sprintf("missing parameter %q %+v", schemeName, req))
```

Comment:
Do we need to write the complete request object in the error? wondering if just writing the URL would be enough? what does this look like?

Reply:
Thank you for the catch! This printing of request object was not supposed to be there; I added it while I was trying to verify my tests.